test/handler/test_swiftlint_handler.vader

```diff
@@ -4,12 +4,14 @@ Execute(The swiftint handler should parse error messages correctly):
   \   {
   \     'lnum': 1,
   \     'col': 7,
+  \     'filename': '<nopath>',
   \     'text': 'Operator Usage Whitespace Violation: Operators should be surrounded by a single whitespace when they are being used. (operator_usage_whitespace)',
   \     'type': 'W',
   \   },
   \   {
   \     'lnum': 1,
   \     'col': 11,
+  \     'filename': '<nopath>',
```

Comment:
These filenames will probably mess things up. What I recommend is checking for "filenames" starting with `'<'` in the callback, and don't set the filenames in those cases. Then ALE will see neither a `'filename'` nor a `'bufnr'` key, and assume that the problem is for the current buffer.

An easy startswith check is `l:x[:0] is# '<'`.

You can read `ale#engine#FixLocList` to see exactly what ALE does.

Reply:
I agree, I just didn't know how to make this work properly/read it.